### DISCUSSION TITLE
subversion stopped working

### DISCUSSION THREAD

**Discussion by rdw4nau** [Homebrew/discussions#4500] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.17-61-g5f2ccd2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5f2ccd22f2ed3b90ab4ebdc52c9aba2b149de1b5
Last commit: 9 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e7bbc68273847775d450f47a5d69da09c961c2cd
Core tap last commit: 6 hours ago
Core tap branch: master
Core tap JSON: 10 May 22:34 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 20
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 20-core 64-bit cometlake
Clang: 14.0.3 build 1403
Git: 2.40.1 => /usr/local/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.0.1-x86_64
CLT: 14.3.0.0.1.1679647830
Xcode: 14.3
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Putting non-prefixed coreutils in your path can cause GMP builds to fail.

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/subversion.rb
```

### Description of issue

subversion stopped working today... super bummed:

❯ svn info
svn: E200029: Couldn't perform atomic initialization
svn: E200030: SQLite compiled for 3.39.5, but running with 3.39.4

❯ brew list --versions sqlite
sqlite 3.41.2
❯ brew list --versions subversion
subversion 1.14.2_2

❯ svn --version --verbose
svn, version 1.14.2 (r1899510)
   compiled May  8 2023, 06:18:44 on x86_64-apple-darwin22.4.0
...
System information:

* running on x86_64-apple-darwin22.1.0
  - macOS 13.0.1, build 22A400
* linked dependencies:
  - APR 1.7.4 (compiled with 1.7.4)
  - APR-Util 1.6.3 (compiled with 1.6.3)
  - Expat 2.4.8 (compiled with 2.5.0)
  - **SQLite 3.39.4 (compiled with 3.39.5)**
  - Utf8proc 2.8.0 (compiled with 2.8.0)
  - ZLib 1.2.11 (compiled with 1.2.11)
  - LZ4 1.9.4 (compiled with 1.9.4)

Did a bunch of uninstall/install of both svn and sqlite to no avail.
It appears that the subversion formula locks to OS version of SQLite.
Modifying subversion formula to use brew's SQLite appears to have resolved at least 'svn info' command.

What should I do? I'm not a fan of formula mods...

**Comment by SMillerDev** [comment, marked as answer]
Update your macOS. Svn compiled against the newer SQLite in macOS 13.3 and you're a lot of security fixes behind that.

**Reply by rdw4nau** [reply]
TYVM. Did not notice that *yikes* MacOS was hanging up on detecting updates until after a reboot was initiated.

**Reply by sailornet** [reply]
its helpful,very  thanks

**Reply by rahulunni73** [reply]
its very helpful, thanks a lot

**Reply by JohnWong** [reply]
It doesn't work for me on macOS Sonoma 14.1.1

**Reply by Jvaeyhcd** [reply]
i have the same problem, do you have solved it? @JohnWong

**Reply by dnesbitt-ta** [reply]
It doesn't work for me on macOS Sonoma 14.1.2

```
$ svn status
svn: E200029: Couldn't perform atomic initialization
svn: E200030: SQLite compiled for 3.43.2, but running with 3.39.5
```

**Reply by vrkansagara** [reply]
https://github.com/Homebrew/discussions/discussions/4500#discussioncomment-7991886

**Reply by dnesbitt-ta** [reply]
vrkansagara, I did `brew update && brew reinstall sqlite svn --build-from-source`, but this is still the output I get:
```
==> Summary
🍺  /usr/local/Cellar/subversion/1.14.3: 247 files, 31.0MB, built in 7 minutes 4 seconds
==> Running `brew cleanup subversion`...
Removing: /Users/dnesbitt/Library/Caches/Homebrew/subversion--patch--35acd6aebc19843f1a2b3a63e880baceb0f5278ab1ace661e57a502d9d78c93c.diff... (960B)
Removing: /Users/dnesbitt/Library/Caches/Homebrew/subversion--py3c--1.4.tar.gz... (47.0KB)
Removing: /Users/dnesbitt/Library/Caches/Homebrew/subversion--serf--1.3.10.tar.bz2... (144.3KB)
==> Upgrading 2 dependents of upgraded formulae:
Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
python 3.11.0 -> 3.11.7, python@3.9 3.9.15 -> 3.9.18_1
==> Downloading https://ghcr.io/v2/homebrew/core/python/3.11/manifests/3.11.7
Already downloaded: /Users/dnesbitt/Library/Caches/Homebrew/downloads/3ea14c4bf50e0fe04bd6aba45093463c7507c4261dbae96cefe33f0af2000404--python@3.11-3.11.7.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/python/3.9/manifests/3.9.18_1
Already downloaded: /Users/dnesbitt/Library/Caches/Homebrew/downloads/688848987f3e2dc6eede07b9376bcee5829e6afcdcf415e2b230e9809acc3ea3--python@3.9-3.9.18_1.bottle_manifest.json
==> Checking for dependents of upgraded formulae...
==> No broken dependents found!
==> Caveats
==> subversion
svntools have been installed to:
  /usr/local/opt/subversion/libexec

The perl bindings are located in various subdirectories of:
  /usr/local/opt/subversion/lib/perl5

Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
$ svn status
svn: E200029: Couldn't perform atomic initialization
svn: E200030: SQLite compiled for 3.43.2, but running with 3.39.5
```

**Reply by vrkansagara** [reply]
@dnesbitt-ta try following
~~~sh
brew update
brew remove sqlite svn
brew reinstall sqlite svn --build-from-source
~~~

**Reply by w962402254** [reply]
I found that the problem was solved after updating the system. My current system version is 14.2.1 (23C71)

**Comment by KaelInvoker** [comment]
I also encountered this issue. May I ask how it was eventually resolved?

**Reply by SMillerDev** [reply]
By updating macOS to a recent version, as was explained in the accepted answer

**Comment by omvaishnav** [comment]
How can I resolve this issue if there's no available update for macOS and the error persists?

**Comment by inii** [comment]
Updated my macOS. A error like that: svn: E170013: Unable to connect to a repository at URL 'https://xxxxxxx/'
svn: E120171: 执行上下文错误: An error occurred during SSL communication.

**Reply by helee9199** [reply]
我也是一模一样的问题， 昨天重装了系统后安装brew  svn操作就会报这个问题 你解决了吗？

**Comment by vrkansagara** [comment]
I was faced the same issue and resolved using bellow command and no need to upgrade OS.

~~~sh
brew update && brew reinstall sqlite svn --build-from-source
~~~

**Reply by Jvaeyhcd** [reply]
@vrkansagara thank you! this problem bothered me for a week!

**Reply by dutts** [reply]
This is what worked for me too, macOS Sonoma 14.1.1

**Reply by arthurFIRE** [reply]
Thank you. I was able to resolve it thanks to you.

**Reply by FedericoBruzzone** [reply]
Works for me on Sonoma 14.1.2 👍

**Reply by sagielightricks** [reply]
How did you build svn? I constantly get `Errno::ENOENT: No such file or directory @ rb_sysopen - /.../Library/Caches/Homebrew/downloads/341faafd274c164961d080ffbaab85a010d9385563dc574212244e60297de7b8--py3c-1.1.tar.gz`
And when I manually downloaded it from `py3c` i get 
```
ld: Undefined symbols:
  _ERR_GET_FUNC, referenced from:
      _ssl_need_client_cert in ssl_buckets.os
clang: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [libserf-1.1.3.0.dylib] Error 1
```
why does nothing ever work?

**Reply by FedericoBruzzone** [reply]
Don't be so pessimistic :D

1. First I suggest updating your operating system
2. `brew update`
3. `brew remove sqlite svn`
4. `brew reinstall sqlite svn --build-from-source`

Make sure it is built from source, it may take some time.

**Reply by sagielightricks** [reply]
Thanks for the quick response :)
I've done all those steps about 100 times in the past 3 hours. The sqlite builds but SVN always fails on the aforementioned errors. Not sure how it's possible that we get different results.
I tried downloading a `tar.gz` of an older SVN build, maybe that will just work?

**Reply by sagielightricks** [reply]
Well nope it did not work. Back to trying to build from source. I really don't understand what I'm missing here. The error is consistent, it is missing this `py3c-1.1.tar.gz` file and when i try to hack my way around it it fails with `Undefined symbols`. Any idea how to approach that?

**Reply by KKChong** [reply]
after i upgraded my mac os to 14.3, is working fine now

**Reply by China-quanda** [reply]
我遇到了这样的问题
svn: E200029: Couldn't perform atomic initialization
svn: E200030: SQLite 编译为 3.43.2，但是运行于 3.39.5

解决方法是brew update && brew reinstall sqlite svn --build-from-source
